api/src/main/java/com/exelaration/abstractmemery/filters/JWTAuthenticationFilter.java

```diff
@@ -0,0 +1,62 @@
+package com.exelaration.abstractmemery.filters;
+
+import static com.auth0.jwt.algorithms.Algorithm.HMAC512;
+import static com.exelaration.abstractmemery.constants.SecurityConstants.EXPIRATION_TIME;
+import static com.exelaration.abstractmemery.constants.SecurityConstants.HEADER_STRING;
+import static com.exelaration.abstractmemery.constants.SecurityConstants.SECRET;
+import static com.exelaration.abstractmemery.constants.SecurityConstants.TOKEN_PREFIX;
+
+import com.auth0.jwt.JWT;
+import com.exelaration.abstractmemery.domains.ApplicationUser;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+public class JWTAuthenticationFilter extends UsernamePasswordAuthenticationFilter {
+
+  @Autowired private AuthenticationManager authenticationManager;
+
+  public JWTAuthenticationFilter(AuthenticationManager authenticationManager) {
```

Comment:
When we didn't have the public constructor for line 31 but were still autowiring on line 29, this didn't work and resulted in a Null Pointer Exception. Could you explain why we still need line 31 if we are autowiring?

Reply:
I am also curious about this since in the tutorial I was following they didn't use the @autowired annotation.  When I tried to add them and remove the constructor it didn't work.